### PR TITLE
Properly propagate runtime when submitting notebook

### LINKED ISSUE #1300: Submit notebook as pipeline fails
**Describe the issue**
See error message

**To Reproduce**
1. Create a simple notebook
2. Configure a KFP runtime environment
3. Submit notebook for execution in that env
4. See error. Note the reference to Airflow even though KFP was selected as runtime

**Screenshots or log output**
If applicable, add screenshots or log output to help explain your problem.
<details><summary>Log Output</summary>
<pre>
[I 2021-02-17 15:26:56.219 ServerApp] airflow 'Untitled-0217152656' - Submitting pipeline 
[I 2021-02-17 15:26:56.220 ServerApp] Creating pipeline definition as a .py file
[I 2021-02-17 15:26:56.232 ServerApp] airflow 'Untitled-0217152656' - processing pipeline dependencies to: http://cloning1.fyre.ibm.com:31671 bucket: pipeline-artifacts folder: Untitled-0217152656 
[I 2021-02-17 15:26:56.251 ServerApp] airflow 'Untitled-0217152656':'Untitled' - processing operation dependencies for id: 46d4d67f-3241-4ae6-87a3-1636f02cfe9b 
[I 2021-02-17 15:26:56.254 ServerApp] airflow 'Untitled-0217152656':'Untitled' - generated dependency archive: /var/folders/h8/l3qh8v192rv_vvggc6wyphzc0000gn/T/elyra/Untitled-46d4d67f-3241-4ae6-87a3-1636f02cfe9b.tar.gz (0.003 secs)
[I 2021-02-17 15:26:56.913 ServerApp] airflow 'Untitled-0217152656':'Untitled' - uploaded dependency archive to: Untitled-0217152656/Untitled-46d4d67f-3241-4ae6-87a3-1636f02cfe9b.tar.gz (0.173 secs)
[I 2021-02-17 15:26:56.913 ServerApp] airflow 'Untitled-0217152656' - pipeline dependencies processed (0.681 secs)
[E 2021-02-17 15:26:56.965 ServerApp] Uncaught exception POST /elyra/pipeline/schedule?1613604416188 (127.0.0.1)
    HTTPServerRequest(protocol='http', host='127.0.0.1:8888', method='POST', uri='/elyra/pipeline/schedule?1613604416188', version='HTTP/1.1', remote_ip='127.0.0.1')
    Traceback (most recent call last):
      File "/opt/anaconda3/envs/elyra_dev/lib/python3.7/site-packages/tornado/web.py", line 1704, in _execute
        result = await result
      File "/opt/anaconda3/envs/elyra_dev/lib/python3.7/site-packages/elyra/pipeline/handlers.py", line 89, in post
        response = await PipelineProcessorManager.instance().process(pipeline)
      File "/opt/anaconda3/envs/elyra_dev/lib/python3.7/site-packages/elyra/pipeline/processor.py", line 75, in process
        res = await asyncio.get_event_loop().run_in_executor(None, processor.process, pipeline)
      File "/opt/anaconda3/envs/elyra_dev/lib/python3.7/concurrent/futures/thread.py", line 57, in run
        result = self.fn(*self.args, **self.kwargs)
      File "/opt/anaconda3/envs/elyra_dev/lib/python3.7/site-packages/elyra/pipeline/processor_airflow.py", line 79, in process
        repo = github_client.get_repo(github_repo)
      File "/opt/anaconda3/envs/elyra_dev/lib/python3.7/site-packages/github/MainClass.py", line 338, in get_repo
        assert isinstance(full_name_or_id, (str, int)), full_name_or_id
    AssertionError: None
</pre>
</details>

**Expected behavior**

**Deployment information**
Describe what you've deployed and how:
 - Elyra version: 2.0.x
 - Operating system: macos
 - Installation source:  from source
 - Deployment type: local installation

**Pipeline runtime environment**
If the issue is related to pipeline execution, identify the environment where the pipeline is executed
 - Kubeflow Pipelines (provide version number, whether multi-user auth enabled)
 - Apache Airflow (provide version number)
 
**Runtime configuration settings**
Not relevant in this scenario

### PR DESCRIPTION
**Pull request by lresende**
Fixes #1300 
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by lresende** [comment]
One thing that we might need to discuss is the name mismatches between the frontend and backend. 

We introduced `runtime platform` on the UI while all the other places are considering it `runtime`. Should we use `runtime` throughout the code?

File: packages/pipeline-editor/src/SubmitNotebookButtonExtension.tsx [hunk 1]
```diff
@@ -71,16 +71,20 @@ export class SubmitNotebookButtonExtension
...
     const {
+      runtime_platform,
       runtime_config,
       framework,
       dependency_include,
       dependencies,
       ...envObject
     } = dialogResult.value;
 
+    console.log(dialogResult.value);
```
Comment (ptitzler) [inline packages/pipeline-editor/src/SubmitNotebookButtonExtension.tsx:82]: Is this debug?

**Comment by karlaspuldaro** [comment]
I was able to successfully run it on airflow, but not on kfp, this is the error I get:
```
Error making request
Error compiling pipeline hello-world-0218163946 for engine None at /var/folders/gz/byw9z3gn1wvddrp1_gt6cg140000gn/T/tmp8k8wzae2/hello-world-0218163946.tar.gz
```
Even after selecting a pipeline engine on my kfp runtime config, I still get the same type of error.

**Comment by ptitzler** [comment]
> I was able to successfully run it on airflow, but not on kfp, this is the error I get:
> 
> ```
> Error making request
> Error compiling pipeline hello-world-0218163946 for engine None at /var/folders/gz/byw9z3gn1wvddrp1_gt6cg140000gn/T/tmp8k8wzae2/hello-world-0218163946.tar.gz
> ```
> 
> Even after selecting a pipeline engine on my kfp runtime config, I still get the same type of error.

What's in the console log?

**Review by ptitzler — APPROVED** [review]
Worked for both runtimes without any issues in my environment.

**Pull request merged**